lib/helpers.js

```diff
@@ -110,11 +110,15 @@ function getJavaHome () {
   throw new Error("JAVA_HOME is not set currently. Please set JAVA_HOME.");
 }
 
-/*
- * Checks mShowingLockscreen in dumpsys output to determine if lock screen is showing
- */
+ /**
+  * Checks mShowingLockscreen or mDreamingLockscreen in dumpsys output to determine
+  * if lock screen is showing
+  *
+  * @param {string} dumpsys - The output of dumpsys window command.
+  * @return {boolean} True if lock screen is showing.
+  */
 function isShowingLockscreen (dumpsys) {
-  let m = /mShowingLockscreen=\w+/gi.exec(dumpsys);
+  let m = /(mShowingLockscreen=\w+|mDreamingLockscreen=\w+)/gi.exec(dumpsys);
```

Comment:
Is it never the case that `mShowingLockScreen` is present (but false) and `mDreamingLockscreen` is true?

Reply:
https://android.googlesource.com/platform/frameworks/base/+/android-6.0.0_r1/services/core/java/com/android/server/policy/PhoneWindowManager.java#4507

Reply:
Does that answer my question?

My concern is that this function only checks the first match, which, if there are both `mShowingLockScreen` and `mDreamingLockscreen` entries, will only be whatever comes first. I'd rather see a check for both entries, and what their values are.

Reply:
I'm not sure that we should skip the value of `mShowingLockScreen` if it's `false` - this can broke current behavior.
I should double check.

Reply:
{mShowingLockScreen: false, mDreamingLockscreen: true} indicates that device starts showing dream (screensaver) from lock screen. So it seems reasonable to catch this case.